src/main/scala/cromwell/engine/backend/local/SharedFileSystem.scala

```diff
@@ -97,28 +108,85 @@ trait LocalFileSystemOperations {
-    case w: WdlFile => stageWdlFile(w, hostInputsPath)
-    case a: WdlArray => stageInputArray(a, hostInputsPath)
-    case x => x
+  private def stageWdlValue(call: Option[Call], value: WdlValue, hostInputsPath: Path): Try[WdlValue] = value match {
+    case w: WdlFile => stageWdlFile(call, w, hostInputsPath)
+    case a: WdlArray => stageInputArray(call, a, hostInputsPath)
+    case x => Success(x)
   }
 
-  private def stageWdlFile(wdlFile: WdlFile, hostInputsPath: Path): WdlFile = {
+  private def stageWdlFile(call: Option[Call], wdlFile: WdlFile, hostInputsPath: Path): Try[WdlFile] = {
     val originalPath = Paths.get(wdlFile.value)
     val executionPath = hostInputsPath.resolve(originalPath.getFileName.toString)
+
+    def invalidLocalization(s: String)(call: Option[Call], originalPath: Path, executionPath: Path): Try[Unit] =
+      Failure(new UnsupportedOperationException(s"Unrecognized localization strategy: $s"))
+
+    val localizers = LocalizationStrategies map {
+      case "hard-link" => localizePathViaHardLink _
+      case "soft-link" => localizePathViaSymbolicLink _
+      case "copy" => localizePathViaCopy _
+      case s => invalidLocalization(s) _
+    }
+
+    val attemptedLocalization = Stream(localizers: _*) map { _(call, originalPath, executionPath) } find { _.isSuccess }
+    attemptedLocalization match {
+      case None => Failure(throw new UnsupportedOperationException(s"Could not localize $wdlFile -> $executionPath"))
+      case _ => Success(WdlFile(executionPath.toString))
+    }
+  }
+
+  private def localizePathViaCopy(call: Option[Call], originalPath: Path, executionPath: Path): Try[Unit] = {
     if (Files.isDirectory(originalPath)) {
-      FileUtils.copyDirectory(originalPath.toFile, executionPath.toFile)
+      Try(FileUtils.copyDirectory(originalPath.toFile, executionPath.toFile))
     } else {
-      FileUtils.copyFile(originalPath.toFile, executionPath.toFile)
+      Try(FileUtils.copyFile(originalPath.toFile, executionPath.toFile))
+    }
+  }
+
+  private def localizePathViaHardLink(call: Option[Call], originalPath: Path, executionPath: Path): Try[Unit] =
+    Try(Files.createLink(executionPath, originalPath))
+
+  /**
+   * TODO: The 'call' parameter here represents the call statement in WDL that references this path.
+   * We're supposed to not use symbolic links if the call uses Docker.  However, this is currently a
+   * bit incorrect because multiple calls can reference the same path if that path is in a declaration.
+   *
+   * The symbolic link will only fail in the Docker case if a Call uses the file directly and not
+   * indirectly through one of its input expressions
+   */
+  private def localizePathViaSymbolicLink(call: Option[Call], originalPath: Path, executionPath: Path): Try[Unit] = {
+    call.map(_.docker) match {
+      case Some(Some(s)) => Failure(new UnsupportedOperationException("Cannot localize with symbolic links with Docker"))
```

Comment:
It should match the behavior in the docs.  If this function fails, it just tries the next one.  The error is silently ignored... maybe it should be logged.